### PR TITLE
@orta: initial commit with braque models and Rails

### PR DESCRIPTION
**Pull request by dylanfareed**
This PR provides a vanilla rails app with no database and rspec/capybara. It uses a really early gem called [Braque](https://github.com/dylanfareed/braque) to wrap the hyperclient calls to the Hypermedia [artsy/echo API](https://github.com/artsy/echo).

Everything basically works with two ENORMOUS caveats
- There is no user authentication. I want to set up a simple open source compatible way to auth via the primary Artsy API. Short of that this is wide open and shouldn't be deployed quite yet. I'll add an issue for basic auth in the meantime.
- There is no styling at all at the moment. Really, barely a thought as to how we will lay this out so that its functional. I would like to take advantage of either of the Artsy style guides for this os projecto. Will issue and follow up shortly.

Other than that we are golden.

### CODE REVIEW

**Comment by dylanfareed** [comment]
I'll make some notes inline as this clocks in at over 100 files and 3k loc.

**Comment by orta** [comment]
but the Braque README doesn't say where the name comes from, how can we trust this?

File: app/models/account.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Account
+  include Braque::Model
+  self.api_root = Rails.application.config_for(:echo_api)['url']
+  self.api_token = Rails.application.config_for(:echo_api)['token']
```
Comment (dylanfareed) [inline app/models/account.rb:4]: [Braque](https://github.com/dylanfareed/braque) provides a thin wrapper around [hyperclient](https://github.com/codegram/hyperclient). This sets up the `Account` model with remote API details and basic model attributes

File: app/controllers/accounts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  def index
+    @accounts = Account.list page: @page, size: @size
+  end
+
+  def new
+  end
+
+  def create
+    @account = Account.create params[:account]
```
Comment (dylanfareed) [inline app/controllers/accounts_controller.rb:13]: Braque also provides a "ActiveRecord"-like interface for doing basic CRUD.

File: config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
+Rails.application.routes.draw do
+  resources :accounts do
+    resources :features, module: :accounts
+    resources :messages, module: :accounts
+    resources :routes, module: :accounts
+  end
+  root 'accounts#index'
```
Comment (dylanfareed) [inline config/routes.rb:7]: Pretty simple app. CRUD for accounts + namespaced features, messages and routes.

It was very very weird actually having a model called "Route" given how richly associated that already is in Rails-land.

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
+  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
+  <%= csrf_meta_tags %>
+</head>
+<body>
+
+<%= yield %>
+
+</body>
+</html>
```
Comment (dylanfareed) [inline app/views/layouts/application.html.erb:14]: Really nothing to see here yet. Just vanilla layout and stylesheets.

File: spec/controllers/accounts/features_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+      end
+      it 'makes the feature delete request' do
+        expect(feature_delete_request).to have_been_requested
+      end
+      it 'makes the account find request' do
+        expect(account_find_request).to have_been_requested
+      end
+      it 'makes the feature find request' do
+        expect(feature_find_request).to have_been_requested
+      end
```
Comment (dylanfareed) [inline spec/controllers/accounts/features_controller_spec.rb:235]: Pretty quick but complete-ish pass at controller specs. There's lots of repetition in setup between various controllers and view specs. Should be refactored down later on.

File: spec/models/route_spec.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    it 'provides the correct .api_token' do
+      expect(Route.api_token).to eq Rails.application.config_for(:echo_api)['token']
+    end
+  end
+  context 'attributes' do
+    it 'should include expected keys' do
+      expect(Route.attributes.keys).to include(
+        'id', 'name', 'path', 'account_id', 'created_at', 'updated_at'
+      )
+    end
```
Comment (dylanfareed) [inline spec/models/route_spec.rb:20]: These model specs actually just verify that the Braque models are set up in an expected manner.

File: spec/views/accounts/new.html.haml_spec.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+RSpec.describe 'accounts/new.html.erb', type: :view do
+  context 'anonymous user' do
+    before do
+      root_request
+      page.visit new_account_url
+    end
+
+    it 'renders a new account form' do
+      expect(find_field('Name').visible?).to eq true
```
Comment (dylanfareed) [inline spec/views/accounts/new.html.haml_spec.rb:11]: Reallly weak sauce view specs. But in the primordial place we are now, its more a placeholder/proof of setup.

File: spec/controllers/accounts/features_controller_spec.rb (hunk 5 above)
Comment (orta) [inline spec/controllers/accounts/features_controller_spec.rb:235]: are all these generated? Or do you set your keyboard on :fire: ?

**Comment by orta** [comment]
Looks wicked sick to me.

**Pull request merged**